### PR TITLE
refactor!(client): replace the optional `RequestConfig` arg to `Client::send()` with a `with_request_config()` builder method

### PR DESCRIPTION
**Pull request by bnjbvr**
Instead of `Client::send(request, request_config)`, consumers can now do `Client::send(request).with_request_config(request_config)`. The parameter is still `Option`al to avoid messy patterns, when the `request_config` is an `Option` passed from the caller.

The messy pattern would look like this, if `with_request_config` took a `RequestConfig` and not an `Option<RequestConfig>`:

```
let mut send_req = client.send(req);
if let Some(request_config) = request_config {
  send_req = send_req.with_request_config(request_config);
}
```

### CODE REVIEW

File: crates/matrix-sdk/src/client/futures.rs [hunk 1]
```diff
@@ -77,6 +77,13 @@ impl<R> SendRequest<R> {
         self
     }
 
+    /// Use the given [`RequestConfig`] for this send request, instead of the
+    /// one provided by default.
+    pub fn with_request_config(mut self, request_config: Option<RequestConfig>) -> Self {
```
Comment (poljar) [inline crates/matrix-sdk/src/client/futures.rs:82]: Why would we take an option here? Not calling this is the same as passing `None`, no?

**Review by poljar — CHANGES_REQUESTED** [review]

File: crates/matrix-sdk/src/client/futures.rs (hunk 1 above)
Comment (bnjbvr) [reply crates/matrix-sdk/src/client/futures.rs:82]: It's to avoid the messy pattern described in OP and commit message. I was a bit torn about this.

Comment (jplatte) [reply crates/matrix-sdk/src/client/futures.rs:82]: Could do `impl Into<Option<RequestConfig>>` so people can pass a `RequestConfig` directly. There's no reason to ever call it with `None` so type inference should not be an issue.

Comment (poljar) [reply crates/matrix-sdk/src/client/futures.rs:82]: That's a good idea.

**Comment by poljar** [comment]
To prevent bitrot on this during the holidays, I took the freedom to implement the change myself, I also added a changelog entry and reworded the now obsolete commit message.

**Review by poljar — APPROVED** [review]

File: crates/matrix-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased] - ReleaseDate
 
+### Refactor
+
+- [**breaking**] Move the optional `RequestConfig` argument of the
+  `Client::send()` method to the `with_request_config()` builder method. You
+  should call `Client::send(request).with_request_config(request_config).awat`
```
Comment (jplatte) [inline crates/matrix-sdk/CHANGELOG.md:13]:
```suggestion
  should call `Client::send(request).with_request_config(request_config).await`
```

Comment (poljar) [reply crates/matrix-sdk/CHANGELOG.md:13]: Oh, that's a nice typo.

Comment (poljar) [reply crates/matrix-sdk/CHANGELOG.md:13]: https://github.com/matrix-org/matrix-rust-sdk/pull/4446

**Pull request merged**